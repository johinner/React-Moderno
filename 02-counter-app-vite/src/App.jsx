import PropTypes from "prop-types"

export function App( {title} ) {
  //console.log(props)
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>Aprendiendo React</p>
    </>
  );
}

const nombre = () => {
  return "johinner desde function"
}

App.propTypes = {
title: PropTypes.string.isRequired
}

App.defaultProps = {
  //title: "no hay titulo"
}