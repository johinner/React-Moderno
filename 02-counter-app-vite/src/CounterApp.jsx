import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleApp = (e) => {
    setCounter(
      e.target.name === "+1"
        ? counter + 1
        : e.target.name === "-1"
        ? counter - 1
        : e.target.name === "0"
        ? value
        : console.error("No encontrado")
    );
  };
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button name="+1" onClick={handleApp}>
        +1
      </button>
      <button name="-1" onClick={handleApp}>
        -1
      </button>
      <button aria-label="btn-reset" name="0" onClick={handleApp}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};
