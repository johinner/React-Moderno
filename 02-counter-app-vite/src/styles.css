button{
    padding: 15px;
    font-size: 1.3rem;
    background-color: rgb(154, 178, 243);
    margin-right: 5px;
}