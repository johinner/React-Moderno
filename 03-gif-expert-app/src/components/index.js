export * from './AddCategory';
export * from './GifGrid';
export * from './GitfItem';
