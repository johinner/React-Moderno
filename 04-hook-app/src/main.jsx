import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'

import { BrowserRouter } from "react-router-dom";

import { TodoApp } from './08-useReducer/TodoApp'
import { MainApp } from './09-useContext/MainApp'

ReactDOM.createRoot(document.getElementById('root')).render(
    <BrowserRouter>
        {/* <React.StrictMode> */}
            <MainApp />
        {/* </React.StrictMode> */}
    </BrowserRouter>
)
