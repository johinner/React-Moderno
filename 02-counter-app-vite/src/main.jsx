import React from 'react'
import ReactDOM from 'react-dom/client'

import "./styles.css"
import {App} from "./App"
import { CounterApp } from './CounterApp'
import { getHeroeByIdAsync } from './base-pruebas/09-promesas'


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <App title = "Pruebas unitarias en React"/> */}
    <CounterApp value = {27}/>
  </React.StrictMode>
)

 