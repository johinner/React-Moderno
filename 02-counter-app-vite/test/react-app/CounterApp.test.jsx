import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../../src/CounterApp";

describe("Prueba en componente <CounterApp />", () => {
  const value = 10;
  //fotocopia del componente en html
  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el valor inicial en 100", () => {
    render(<CounterApp value={100} />);
    // screen.debug()
    expect(screen.getByText(100)).toBeTruthy();
  });

  test("debe de incrementar con el boton +1", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      `${value + 1}`
    );
  });

  test("debe de decrementar con el boton -1", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("-1"));
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      `${value - 1}`
    );
  });

  test("boton de reset debe devolver a el valor defaul", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("-1"));
    fireEvent.click(screen.getByText("-1"));
    screen.debug();
    fireEvent.click(screen.getByRole("button", { name: "btn-reset" }));
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      `${value}`
    );
  });
});
