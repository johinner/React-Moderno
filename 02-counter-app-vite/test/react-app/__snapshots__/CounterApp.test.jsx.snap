// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    name="+1"
  >
    +1
  </button>
  <button
    name="-1"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
    name="0"
  >
    Reset
  </button>
</div>
`;
